### ISSUE TITLE
Crash of web site when call two items with the same name and it has the same children.

### ISSUE THREAD

**Issue by MartinBodocky** [angular-gantt/angular-gantt#517]
I have found bug when you feed gantt with this data:
               [{
                    name: 'SGTS', children: ['Department 1']
                },
                {
                    name: 'Department 1', children: ['Darrell Quirion']
                },
                {
                    name: 'Department 1', children: ['Darrell Quirion']
                }
                ,
                {
                    name: 'Darrell Quirion', tasks: [
                        { name: 'Iran', color: colours[0], from: setNewDate(1), to: setNewDate(4) }
                    ]
                }];
I cannot get any exception or something like that, because the browser just stuck. I'm using Chrome version 27.0.1453.110.

**Comment by Toilal** [comment]
As a workaround, you can introduce `id` property in your model, and use those ids to reference children. This should works (IDs must all be different).

**Comment by Toilal** [comment]
And Darrell Quirion can't have multiple parent ...

**Comment by MartinBodocky** [comment]
Both your comments are right, but the control cannot cause crash of browser :)
The acceptable behavior is display nothing or some error message, for example your data are not valid. What do you think?

**Comment by Toilal** [comment]
Yes, we should raise in error in case there's a loop in Hierarchy building, or duplicates ... Maybe you could open a pull request ? it should be in [Hierarchy factory](https://github.com/angular-gantt/angular-gantt/blob/master/src/core/logic/util/hierarchy.factory.js) ...
